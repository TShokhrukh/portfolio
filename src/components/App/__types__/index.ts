export * from './App.types'
export * from './App.model.types'
