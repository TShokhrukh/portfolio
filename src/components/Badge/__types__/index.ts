import type { ReactNode, ElementType } from 'react'
import type { IClassNameProps } from '@bem-react/core'

export interface IBadgeProps extends IClassNameProps {
  role?: string,
  id?: string,
  component?: ElementType,
  children?: ReactNode
}

export interface IBadgeColorProps {
  color?: 'purple'
}
