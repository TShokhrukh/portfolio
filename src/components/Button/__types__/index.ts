import type { ElementType, ComponentType, ReactNode, SyntheticEvent } from 'react'
import type { IClassNameProps } from '@bem-react/core'

export interface IIconProps {
  className: string
}

export interface IButtonProps extends IClassNameProps {
  component?: ElementType,
  startIcon?: ComponentType<IIconProps>,
  endIcon?: ComponentType<IIconProps>,
  rounded?: boolean,
  disabled?: boolean,
  type?: 'submit'|'button'|'reset',
  id?: string,
  children?: ReactNode,
  onClick?: (event: SyntheticEvent<HTMLButtonElement>) => void
}

export interface IButtonVariantLinkProps {
  variant?: 'link',
  type?: undefined,
  href?: string
}

export interface IButtonIconProps {
  icon: ComponentType<IIconProps>,
  direction: 'start'|'end'
}
