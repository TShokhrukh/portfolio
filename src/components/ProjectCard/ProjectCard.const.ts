import { createContext } from 'react'
import { cn } from '@bem-react/classname'
import type { IProjectCardContext } from './__types__'

export const cnProjectCard = cn('ProjectCard')

export const Context = createContext<IProjectCardContext>(null!)
