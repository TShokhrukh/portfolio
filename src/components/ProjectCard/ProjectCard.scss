.ProjectCard {
  cursor: pointer;
  display: grid;
  grid-template-rows: auto 100px auto 200px;
  &:hover {
    color: inherit;
  }
}
