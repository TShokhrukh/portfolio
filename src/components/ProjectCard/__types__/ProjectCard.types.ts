import type { IClassNameProps } from '@bem-react/core'
import type { IComponentProp } from '~/src/types/react'
import type { TLocale } from '~/src/types'

export interface IProject {
  id: string,
  name: string,
  desc: TLocale<string>,
  stack: string[],
  image: string,
  href?: string,
}

export interface IProjectCardContext {
  project: IProject
}

export interface IProjectCardProps extends IClassNameProps, IComponentProp {
  project: IProject
}
