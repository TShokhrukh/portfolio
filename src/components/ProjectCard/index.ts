export * from './build'
export * from './__types__'
export { ProjectCard as ProjectCardBase } from './ProjectCard'
