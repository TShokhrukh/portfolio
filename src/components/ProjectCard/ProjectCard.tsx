import React, { FC } from 'react'
import { useTranslator } from '@eo-locale/react'
import { Card } from '~/components/Card'
import { Link } from '~/components/Link'
import { Typography } from '~/components/Typography'
import { Context, cnProjectCard } from './ProjectCard.const'
import { ProjectCardStacks } from './Stacks/ProjectCard-Stacks'
import { ProjectCardImg } from './Img'
import type { IProjectCardProps } from './__types__'
import './ProjectCard.scss'

export const ProjectCard: FC<IProjectCardProps> = ({
  project,
  ...props
}) => {
  const t = useTranslator()
  return (
    <Context.Provider value={{ project }}>
      <Card {...props} component={(props) => <Link {...props} href={project.href} target='_blank' rel='noopener noreferrer'/>} className={cnProjectCard({}, [])}>
        <Typography color='white'>{project.name}</Typography>
        <Typography className={cnProjectCard('Desc', [])}>{project.desc[t.language]}</Typography>
        <ProjectCardStacks />
        <ProjectCardImg />
      </Card>
    </Context.Provider>
  )
}
