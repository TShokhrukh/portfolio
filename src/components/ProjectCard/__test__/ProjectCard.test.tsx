/* global test, expect, describe */
import React from 'react'
import { render, screen } from '@testing-library/react'
import { ProjectCard } from '../ProjectCard'
import { IProject } from '../__types__'

describe('ProjectCard', () => {
  test('must return the component template', () => {
    const project: IProject = {
      id: '1',
      href: 'http://localhost:9000/project_card',
      desc: { en: 'project_card' },
      image: 'http://localhost:9000/image.png',
      name: 'project name',
      stack: ['TypeScript', 'JavaScript']
    }
    render(<ProjectCard project={project} data-testid='project_card' />)

    expect(screen.getByTestId('project_card')).toMatchSnapshot()
  })
})
