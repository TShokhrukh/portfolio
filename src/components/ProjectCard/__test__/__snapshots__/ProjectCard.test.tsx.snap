// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProjectCard must return the component template 1`] = `
<a
  class="Link Card ProjectCard"
  data-testid="project_card"
  href="http://localhost:9000/project_card"
  rel="noopener noreferrer"
  target="_blank"
>
  <p
    class="Typography Typography_color_white"
  >
    project name
  </p>
  <p
    class="Typography ProjectCard-Desc"
  >
    project_card
  </p>
  <div
    class="ProjectCard-Stacks"
  >
    <span
      class="Badge Badge_color_purple"
    >
      TypeScript
    </span>
    <span
      class="Badge Badge_color_purple"
    >
      JavaScript
    </span>
  </div>
  <div
    class="ProjectCard-Image"
  >
    <img
      alt="project name"
      class="Img Img_fill_horizontal"
    />
  </div>
</a>
`;
